capstone-frontend/src/components/ClubPage.jsx

```diff
@@ -1,10 +1,34 @@
 import React, { Component } from 'react';
 
 class ClubPage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      club: {}
+    }
+  }
+
+  fetchClubData = () => {
+    fetch(`/api/clubs?id=${this.props.match.params.id}`)
+        .then(response => response.json()).then(res => this.setState({club: res[0]}));
+  }
+
+  componentDidMount() {
+    this.fetchClubData();
+  }
+
+  componentDidUpdate() {
+    this.fetchClubData();
```

Comment:
I think you'll want to check that the relevant prop changed before calling this, otherwise you'll end up in the same kind of render loop. [`componentDidUpdate`](https://reactjs.org/docs/react-component.html#componentdidupdate) gives you access to `prevProps` for this purpose.

Reply:
I didn't realize that, but that makes a lot of sense. Changing this to check for a change in state before calling fetchData.